src/libMediator/Mediator.cpp

```diff
@@ -118,3 +119,44 @@ std::string Mediator::GetNodeMode(const Peer& peer)
         return "SHRD";
     }
 }
+
+void Mediator::HeartBeat_Init()
+{
+#ifndef IS_LOOKUP_NODE
+    LOG_MARKER();
+    m_heartBeatTime = 0;
+
+    auto func = [this]() -> void {
+        const unsigned int heartBeatTimeout = (2 * 60 * NUM_FINAL_BLOCK_PER_POW)
+            + POW1_WINDOW_IN_SECONDS + BACKUP_POW2_WINDOW_IN_SECONDS
+            + HEARTBEAT_DELTA;
+
+        while (true)
+        {
+            this_thread::sleep_for(chrono::seconds(HEARTBEAT_INTERVAL));
+            lock_guard<mutex> guard(m_heartBeatMutex);
+            m_heartBeatTime += HEARTBEAT_INTERVAL;
+
+            if (m_heartBeatTime < heartBeatTimeout)
+            {
+                LOG_GENERAL(INFO, "Still alive...");
+                continue;
+            }
+
+            LOG_GENERAL(WARNING, "I am DEAD, rejoin to network");
+            (DirectoryService::Mode::IDLE == m_ds->m_mode)
+                ? m_node->RejoinAsNormal()
+                : m_ds->RejoinAsDS();
```

Comment:
we cannot let a DS to rejoin as DS under this heartBeatTimeout, consider when this DS is going to be kicked out in the next DS epoch.

Reply:
So we may let the DS rejoin as a normal node, and inform everyone he's no longer a DS when he tries to rejoin